### PR TITLE
doc/services/llext: Mention the LLEXT EDK

### PR DESCRIPTION
**Pull request by edersondisouza**
Mention the EDK existence, the build target and discuss some features.

### CODE REVIEW

**Review by teburd — DISMISSED** [review]

**Review by dapperlo — DISMISSED** [review]

File: doc/services/llext/index.rst [hunk 1]
```diff
@@ -18,6 +18,67 @@ needed by the :c:func:`llext_load` function. An implementation over a buffer
...
+of different use cases, such as when building mocks for unit tests. The provided
+flags are:
+
+- ``LLEXT_CFLAGS``: Flags needed to build an extension.
+- ``LLEXT_ALL_INCLUDE_CFLAGS``: Include flags needed to include all headers.
+- ``LLEXT_INCLUDE_CFLAGS``: Include flags for non build generated headers.
+- ``LLEXT_GENERATED_INCLUDE_CFLAGS``: Include flags for build generated headers.
+- ``LLEXT_BASE_CFLAGS``: All other flags needed to build an extension,
+  excluding the include flags.
+- ``LLEXT_GENERATED_IMACROS_CFLAGS``: Include only `-imacros` flags for build
```
Comment (kartben) [inline doc/services/llext/index.rst:60]:
```suggestion
- ``LLEXT_GENERATED_IMACROS_CFLAGS``: Include only ``-imacros`` flags for build
```

File: doc/services/llext/index.rst [hunk 2]
```diff
@@ -18,6 +18,67 @@ needed by the :c:func:`llext_load` function. An implementation over a buffer
...
+LLEXT EDK Kconfig options
+-------------------------
+
+The LLEXT EDK can be configured using the following Kconfig options:
+
+:kconfig:option:`CONFIG_LLEXT_EDK_NAME`
+    The name of the generated EDK tarball.
+
+:kconfig:option:`CONFIG_LLEXT_EDK_USERSPACE_ONLY`
+    If set, the EDK will include headers which do not contain code to route
```
Comment (kartben) [inline doc/services/llext/index.rst:72]:
```suggestion
    If set, the EDK will include headers that do not contain code to route
```

**Review by kartben — CHANGES_REQUESTED** [review]
relatively minor grammar/ReStructuredText comments -- thanks for adding this @edersondisouza!

**Comment by edersondisouza** [comment]
v2;
  - Wording and rst fixes.

**Review by kartben — APPROVED** [review]

**Review by dapperlo — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**